### DISCUSSION TITLE
Mission control dashboard

### DISCUSSION THREAD

**Discussion by sandstrom** [bensheldon/good_job#1340] [Ideas]
Would it make sense for Good Job to piggy-back on the Mission Control dashboard project?

https://github.com/rails/mission_control-jobs

So basically write an adapter for it, and then not have to maintain it's own dashboard?

**Comment by bensheldon** [comment]
I haven't used Mission Control. Do you have a sense of difference in features with GoodJob's Dashboard? https://goodjob-demo.herokuapp.com/

I did recently share this desire: https://ruby.social/@bensheldon/112299337515407264

I suspect Mission Control offers a lot less functionality than the GoodJob Dashboard. I'm not opposed to _also_ having an adapter, but I imagine that there is a lot in GoodJob (and future things to build) that isn't supported in Mission Control.

**Reply by sandstrom** [reply]
Good points!

I don't know enough about either of them to compare.

Just wanted to throw the idea out there (by sharing the work, each project could free more time to queue-specifics). Maybe the adapters can have optional properties, such that each queue doesn't have to support/display all types of information in the dashboard.

All that said, not saying this is a good idea, just wanted to raise it 😄

**Reply by bensheldon** [reply]
I'm glad you raised the question! I'm definitely not opposed to maintaining something like that, and it might be beneficial to design for a shared interface. I just probably won't get to it myself unless someone else works on it first.

Also, I hope Solid Queue has helped reinvigorate Active Job. The topic of Mission Control reminds me of this older issue in regards to like "how does new functionality happen?" https://github.com/bensheldon/good_job/issues/255
